app/views/do.py

```diff
@@ -2661,24 +2661,35 @@ def delete_comment():
             comment = SubPostComment.get(SubPostComment.cid == form.cid.data)
         except SubPostComment.DoesNotExist:
             return jsonify(status="error", error=_("Comment does not exist"))
-        post = SubPost.get(SubPost.pid == comment.pid)
+        sub = (
+            SubPost.select(SubPost.sid, Sub.name)
+            .join(Sub)
+            .where(SubPost.pid == comment.pid)
+            .dicts()
+        )[0]
```

Comment:
It's probably more elegant to do a 

```suggestion
        sub = SubPost.select(SubPost.sid, Sub.name)
            .join(Sub)
            .where(SubPost.pid == comment.pid)
            .get()
```

Reply:
That version doesn't work, but swapping the models between the `select` and the `join` does.